src/Sentry/SentrySdk.cs

```diff
@@ -21,20 +21,68 @@ namespace Sentry
     /// </remarks>
     public static class SentrySdk
     {
-        private static IHub _hub = DisabledHub.Instance;
+
+        /// <summary>
+        ///  Holds Hubs per thread or only mainHub if globalHubMode is disabled.
+        /// </summary>
+        private static readonly AsyncLocal<IHub> _currentHub = new AsyncLocal<IHub>();
+        /// <summary>
+        /// The Main Hub or NoOp if Sentry is disabled.
+        /// </summary>
+        private static IHub _mainHud = DisabledHub.Instance;
+
+        /// <summary>
+        /// Default value for globalHubMode is false.
+        /// </summary>
+        private const bool _globalHudDefaultMode = false;
+
+        /// <summary>
+        /// whether to use a single (global) Hub as opposed to one per thread.
+        /// </summary>
+        private static bool _globalHudMode = _globalHudDefaultMode;
```

Comment:
In case other threads are calling methods from this class before `Init` is called, and value is cached

```suggestion
        private static volatile bool _globalHubMode = _globalHubDefaultMode;
```